homeassistant/components/srp_energy/sensor.py

```diff
@@ -94,62 +109,61 @@ def __init__(self, coordinator):
         self._state = None
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of the sensor."""
         return f"{DEFAULT_NAME} {self._name}"
 
     @property
-    def unique_id(self):
+    def unique_id(self) -> str:
         """Return sensor unique_id."""
         return self.type
 
     @property
-    def native_value(self):
+    def native_value(self) -> StateType:
         """Return the state of the device."""
         if self._state:
             return f"{self._state:.2f}"
         return None
 
     @property
-    def native_unit_of_measurement(self):
+    def native_unit_of_measurement(self) -> str:
         """Return the unit of measurement of this entity, if any."""
         return self._unit_of_measurement
 
     @property
-    def icon(self):
+    def icon(self) -> str:
         """Return icon."""
         return ICON
 
     @property
-    def usage(self):
+    def usage(self) -> str:
         """Return entity state."""
+        usage: float = 0.0
         if self.coordinator.data:
-            return f"{self.coordinator.data:.2f}"
-        return None
+            usage = self.coordinator.data
+        return f"{usage:.2f}"
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, str]:
         """Return the state attributes."""
-        if not self.coordinator.data:
-            return None
         attributes = {
             ATTR_ATTRIBUTION: ATTRIBUTION,
         }
 
         return attributes
 
     @property
-    def available(self):
+    def available(self) -> bool:
```

Comment:
Inherit from `CoordinatorEntity` instead. It already has many of these things implemented.

Reply:
That is part on the next pr